### DISCUSSION TITLE
Swiper import error in ssr mode

### DISCUSSION THREAD

**Discussion by sadeq-qafari** [quasarframework/quasar#13160] [CLI - SSR mode]
im trying to use swiper in my quasar v2 codebase
it works fine in spa mode but i get this error in ssr mode:
`Must use import to load ES Module: D:\MY_CODE_BASE\node_modules\swiper\vue\swiper-vue.js require() of ES modules is not supported. require() of D:\MY_CODE_BASE\node_modules\swiper\vue\swiper-vue.js from D:\MY_CODE_BASE is an ES module file as it is a .js file whose nearest parent package.json contains "type": "module" which defines all .js files in that package scope as ES modules. Instead rename swiper-vue.js to end in .cjs, change the requiring code to use import(), or remove "type": "module" from D:\MY_CODE_BASE\node_modules\swiper\package.json.`

**Comment by rstoenescu** [comment]
Swiper is not built with SSR in mind and it must run only on client-side, so do just that.

**Reply by sadeq-qafari** [reply]
Thank you for replying. Are there any instructions that you can reference witch helps me doing that?
I tried using q-no-ssr but it doesn't help.

**Reply by MilosPaunovic** [reply]
If you're instantiating it through boot file, you can add the `server: false` to run only on client side.

[https://quasar.dev/quasar-cli-webpack/boot-files#:~:text=When%20building%20a%20SSR%20app%2C%20you%20may%20want%20some%20boot%20files%20to%20run%20only%20on%20the%20server%20or%20only%20on%20the%20client%2C%20in%20which%20case%20you%20can%20do%20so%20like%20below%3A](https://quasar.dev/quasar-cli-webpack/boot-files#:~:text=When%20building%20a%20SSR%20app%2C%20you%20may%20want%20some%20boot%20files%20to%20run%20only%20on%20the%20server%20or%20only%20on%20the%20client%2C%20in%20which%20case%20you%20can%20do%20so%20like%20below%3A)

**Reply by sadeq-qafari** [reply]
I defined swiper components globally in client-side boot file and it worked!
Thank You!
